src/CModel.cc

```diff
@@ -878,12 +878,15 @@ class CModelAlgorithm::Impl {
             model, fixed, expData.fitSys, expData.position,
             exposure, footprint, expData.psf, UnitTransformedLikelihoodControl(false)
         );
-        ndarray::Array<Pixel,2,-1> modelMatrix = makeModelMatrix(likelihood, nonlinear);
-        Vector gradient = -(modelMatrix.asEigen().adjoint() *
-            likelihood.getUnweightedData().asEigen()).cast<Scalar>();
+        auto unweightedData = likelihood.getUnweightedData();
```

Comment:
I would have to keep the ndarray around as well as the Eigen::Map view of it (since the former owns the memory) and I'm not very happy keeping track of two variables that contain different views of the same data.